File: Leetcode/3663 - Find The Least Frequent Digit/solution.py
def getLeastFrequentDigit(n: int) -> int:
    digFreq = [0] * 10
    temp = n
    while temp > 0:
        digFreq[temp % 10] += 1
        temp //= 10

    leastFreqNum = -1
    for i in range(10):
        if digFreq[i] == 0:
            continue
        if leastFreqNum == -1 or digFreq[i] < digFreq[leastFreqNum]:
            leastFreqNum = i
    return leastFreqNum

File: Leetcode/3663 - Find The Least Frequent Digit/solution.js
var getLeastFrequentDigit = function(n) {
    const digFreq = Array(10).fill(0);
    while (n > 0) {
        digFreq[n % 10]++;
        n = Math.floor(n / 10);
    }

    let leastFreqNum = -1;
    for (let i = 0; i < 10; i++) {
        if (digFreq[i] === 0) continue;
        if (leastFreqNum === -1 || digFreq[i] < digFreq[leastFreqNum]) {
            leastFreqNum = i;
        }
    }
    return leastFreqNum;
};